src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTaggerProvider.cs

```diff
@@ -16,25 +18,34 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
     /// The provider that is used as a middleman to create the tagger so that the data tag
     /// can be used to create the UI tag
     /// </summary>
-    [Export(typeof(ITaggerProvider))]
+    [Export(typeof(IViewTaggerProvider))]
     [ContentType(ContentTypeNames.RoslynContentType)]
     [TagType(typeof(IntraTextAdornmentTag))]
     [Name(nameof(InlineParameterNameHintsTaggerProvider))]
-    internal class InlineParameterNameHintsTaggerProvider : ITaggerProvider
+    internal class InlineParameterNameHintsTaggerProvider : IViewTaggerProvider
     {
         private readonly IBufferTagAggregatorFactoryService _bufferTagAggregatorFactoryService;
+        public readonly IClassificationFormatMapService ClassificationFormatMapService;
+        public readonly IClassificationTypeRegistryService ClassificationTypeRegistryService;
+        public readonly IThreadingContext ThreadingContext;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public InlineParameterNameHintsTaggerProvider(IBufferTagAggregatorFactoryService bufferTagAggregatorFactoryService)
+        public InlineParameterNameHintsTaggerProvider(IBufferTagAggregatorFactoryService bufferTagAggregatorFactoryService,
+                                                       IClassificationFormatMapService classificationFormatMapService,
+                                                       IClassificationTypeRegistryService classificationTypeRegistryService,
+                                                       IThreadingContext threadingContext)
```

Comment:
It's odd to me that these are imported, but not used in this class. I would expect these to be passed to the constructor of `InlineParameterNameHintsTagger` below, rather than have it access properties (or fields) from this class. It makes the dependencies clearer, and makes mocking easier (though thats not relevant here). If you think about it, the `InlineParameterNameHintsTagger` doesn't actually need an instance of this class at all.